lib/crypt-static.c

```diff
@@ -25,8 +25,17 @@
 char *
 crypt (const char *key, const char *setting)
 {
-  static struct crypt_data nr_crypt_ctx;
-  return crypt_r (key, setting, &nr_crypt_ctx);
+  static TLS char output[CRYPT_OUTPUT_SIZE];
+  struct crypt_data nr_crypt_ctx;
+
+  crypt_r (key, setting, &nr_crypt_ctx);
```

Comment:
Now this calls `crypt_r` with uninitialized stack data in `nr_crypt_ctx` - a bug? IIRC, at least the `initialized` field should be set to 0.

Also, we probably incur some performance penalty by having this struct recreated each time, but perhaps only for `descrypt`?

Reply:
> Now this calls `crypt_r` with uninitialized stack data in `nr_crypt_ctx` - a bug? IIRC, at least the `initialized` field should be set to 0.

Fixed by initializing the whole struct with `explicit_bzero`.


> Also, we probably incur some performance penalty by having this struct recreated each time, but perhaps only for `descrypt`?

Well, the performance penalty is meassurable, but not noticeable:  This change adds on average about 220 milliseconds of computing time for 1 Million invocations; so in a range of 200 - 300 nanoseconds per call.  As those numbers are measured on an Intel Celeron 4205U with Fedora Workstation 41, the real life time penalty on affordable faster hardware, like a Ryzen 5 2400G, are expected be lower.

Reply:
> Fixed by initializing the whole struct with `explicit_bzero`.

Feels like mild misuse of `explicit_bzero`, but that's not necessarily bad. If we want to use it not only to initialize, but also to wipe any sensitive data that might have been left from a previous call, then perhaps we also want to do it for the output buffer before the `strcpy_or_abort`. And then maybe move the `explicit_bzero` of the data struct to after the `strcpy_or_abort` and introduce simple initialization (just `memset`)?

Reply:
> Feels like mild misuse of `explicit_bzero`, but that's not necessarily bad.

Changed to `memset` for initialization.